Fixes towards getting us running on docker-machine.

mindhog opened the pull request:
@aschults here are a set of CLs, one is a minor formatting fix, the other are a set of changes towards getting us to function smoothly on docker-machine with the latest docker version.

mindhog committed:
Fixed incorrect indentation.

Change-Id: I6676da603d3bf6f2f95c6734860eca6419fa1b53

mindhog committed:
Deal with container shutdown race condition.

Newer versions of Docker appear to produce a NullResource exception instead of
an APIError when a container is referenced after it's been shut down, so check
for this as well.

Change-Id: I2f478637e54acddd01f70564f8ccba11bdfcaf08

mindhog committed:
Check for versions between 1.5.0 and 1.8.x.

Change-Id: Ifbc53497f4af1d4af7fcf3215b4602931892c2b3

mindhog committed:
Add information on docker-machine.

Fix references to boot2docker to prefer references to docker-machine
(boot2docker is deprecated in favor of the later).

Change-Id: I3e9fa40b73fd941981c779b4eaf82547262e22e4

mindhog committed:
Fix bad docker version in test.

Make the "bad docker version" 1.4, so it's before the range that we check for.

Change-Id: Ifa131b84085088caf5f7078d0aafa86aae945050

JustinBeckwith commented on README.md:
```diff
@@ -19,13 +19,16 @@ From its root directory, appstart can be installed like so:
-variables. Appstart is known to work for Docker server version 1.5.0.  For
-information about installing docker and/or Boot2Docker see:
+Appstart requires a running Docker server. The server can be running in a 
+docker-machine instance or even a Boot2Docker VM, as long as the docker 
```
Let's not even mention boot2docker at this stage.

JustinBeckwith commented:
LGTM

mindhog committed:
Fixed docker instructions.

Removed references to boot2docker and fixed the docker versions to correspond
to the latest code (the latest docker-py only works against the latest API
version).

Change-Id: I17ab8e632d844f3750577771bfc886b8728e7e0f

mindhog commented:
@JustinBeckwith I've removed the boot2docker references and updated the incorrect version info.
